### ISSUE TITLE
cpp-multithread not working on 32-bit Windows

### ISSUE THREAD

**Issue by kainjow** [andlabs/libui#161]
I tested the cpp-multithread example and it is not working on 32-bit Windows in VS2015, but when I build 64-bit (with `-G "Visual Studio 14 Win64"`) it works. Looks like it's hanging at `cv.wait_for()`.

**Comment by kainjow** [comment]
Ok, actually it's exiting because of `cv_status::no_timeout`. I think this is because of spurious wakeup (?). A flag should probably be kept for if it's quitting or not.

**Comment by andlabs** [comment]
I am probably invoking undefined behavior; I have done this before with the same program as well. Someone who knows how C++11 threads works better than I do will probably need to comb through the code to see.

**Comment by kainjow** [comment]
Should add uiTimer :)

**Comment by andlabs** [comment]
Yes, that will come in the future too =P Would still like an example, no matter how contrived, of how to mix libui and threads; it really shouldn't be too hard *in general*, but I'm not an expert with C++ threads, so I may have gotten *the specifics* wrong.
